### ISSUE TITLE
Recovery from DMM popup

### ISSUE THREAD

**Issue by mrmin123** [mrmin123/kancolle-auto#148]
#### Issue

Multiple people (myself included) have been running into a DMM popup that would cease kancolle-auto's ability to click in the game's screen.

The popup in question:
![](http://i.imgur.com/BP0fhfn.png)
![](https://dl.dropboxusercontent.com/u/73564551/Games/KanColle/error.png)

This popup, per discussion in [this reddit thread](https://www.reddit.com/r/kancolle/comments/4a5lwp/question_why_do_i_sometimes_get_this_dmm_browser/) seems to be independent of kancolle-auto's behavior and could have something to do with DMM (patching of game/API refresh/bot-detect/etc) or the user's connection, or both.

On first glance, recovery from this presents a number of challenges and/or considerations (rambling ahead):
1. As seen from the above screenshots, depending on the user's OS, Viewer, and system settings the style of the window is completely different, thus matching on the popup window itself is not possible unless the user generates their own popup screen capture to match on (may not be a bad idea)
2. Does it allow for the page to refresh, or no? Ideally, if the popup occurs while being in the main screen, the refresh happens, and kancolle-auto goes through the recovery process from game start. If during combat, the refresh would ideally not happen. Maybe force a refresh after?
3. Actually, refreshing regardless of context would actually be the most easy to implement. Otherwise the ability to retry the failed command on Exception would have to be implemented. Perhaps via a retry decorator?

Extra note: that is actually not the only popup I've seen that would cause kancolle-auto to freeze up. As a KC3Kai user, sometimes updates from the plugin would attempt to force a refresh, which would cause the 'Are you sure you'd like to abandon your shipgirls etc' window to popup. Other times, I've seen the page attempt to refresh by itself, with no warning from DMM and no update from KC3Kai.

**Comment by mintgreenrose** [comment]
Can confirm popup appears in the middle of combat. I regret I wasn't able to take a screenshot though, was too eager to close it.

**Comment by mrmin123** [comment]
Is anyone getting this in other non-Chrome viewers/browsers?

**Comment by n0k0m3** [comment]
Nope, used kctool and no error
Btw, this error is from DMM not API request. Got the same error with
MonMusu as well. Probably a DMM solution for NEET <(")

**Comment by silfumus** [comment]
Had those pop-up when playing manually with EO, but not when I was using kancolle-auto.

**Comment by mrmin123** [comment]
@silfumus If you can, could you grab a screenshot of the popup in EO the next time you get it? I'd appreciate it.

**Comment by silfumus** [comment]
![untitled-1](https://cloud.githubusercontent.com/assets/6127734/14068693/52b5990c-f4c6-11e5-892b-5c00ec087415.png)

**Comment by mintgreenrose** [comment]
Hey guys something you might find useful:

http://himeuta.org/showthread.php?7333-Bot-detection-cat-errors-201-cat

**Comment by mrmin123** [comment]
I haven't been getting the DMM popups since a week or so ago, but I also don't use EO so I don't know if the DMM popups equate to the 201 error. My impression was that the DMM popup is different from the anti-bot catbombs, although it's entirely possible that the DMM popup is a new implementation of an anti-bot mechanism... Either way, thanks for the link.

**Comment by silfumus** [comment]
Haven't got DMM pop-up in a while so I couldn't confirm.

I think the DMM pop-up and anti-bot catbomb are two different thing since the former is done by DMM itself and not KanColle Staff. If you had frequent DMM pop-up and/or catbomb, get a better connection.

The only time I got catbomb aside from connection-related issues was when I went too fast clicking the resupply and PVP selection screen. I would be surprised if sikuli managed to pull this off.

Currently ranking for the top 100 but I wonder if KanColle Staff could just ban someone from ranking instead of the game altogether (I haven't received the top 500 February ranking reward, I hope this wasn't the reason).
